app/src/main/java/acr/browser/lightning/adblock/AbpBlocker.kt

```diff
@@ -60,23 +62,21 @@ class AbpBlocker @Inject constructor(
     private val dummyResponse by lazy { WebResourceResponse("text/plain", "UTF-8", EmptyInputStream()) }
 
     init {
-        // TODO: ideally we should call loadLists here (blocking) if the url in current tab (on opening browser) is not special
-        //  because loadLists() here is sometimes significantly faster than inside the GlobalScope
-        //  but generally no need to block UI, better just delay the first web requests
-        //  -> so, how to get current tab url?
-
-        GlobalScope.launch(Dispatchers.Default) {
-            // load lists here if not loaded above
-            //  2-5x slower than blocking for some reason -> is there any reasonable compromise?
-            loadLists()
-
-            // update all enabled entities/blocklists
-            // may take a while depending on how many lists need update, and on internet connection
-            if (abpListUpdater.updateAll(false)) { // returns true if anything was updated
-                removeJointLists()
-                loadLists() // update again if files have changed
+        // hilt always loads blocker, even if not used
+        //  thus we load the lists only if blocker is actually enabled
+        if (userPreferences.adBlockEnabled)
+            GlobalScope.launch(Dispatchers.Default) {
```

Comment:
Only now realized you are using coroutines. Cool stuff, I can learn from you 😁 I wonder if we should make sure it can be cancelled, since it's a network request.

Reply:
So something like using `abpBlockerScope = CoroutineScope(SupervisorJob())` instead of `GlobalScope`?
It could then also be used in `AdBlockSettingsFragment`

Reply:
Maybe, I'm not sure if there is a standard scope that is cancelled when the app exits.
The viewModelScope is notably cancelled when the ViewModel is cleared.
Maybe there is a scope for the app too. I'm new to this.
https://developer.android.com/topic/libraries/architecture/coroutines